src/frontend/data/stores/useLivePanelState/index.spec.ts

```diff
@@ -0,0 +1,278 @@
+import { LivePanelDetail, useLivePanelState } from '.';
+
+describe('useLivePanelState()', () => {
+  it('init with default values', () => {
+    const state = useLivePanelState.getState();
+
+    expect(state.availableDetails).toEqual([]);
+    expect(state.currentDetail).toEqual(undefined);
+    expect(state.isPanelVisible).toBeFalsy();
+  });
+});
+
+describe('state.setAvailableDetails()', () => {
+  it('without changing available details has no side effect', () => {
+    useLivePanelState.setState({
+      isPanelVisible: true,
+      currentDetail: LivePanelDetail.CHAT,
+      availableDetails: [LivePanelDetail.APPLICATION, LivePanelDetail.CHAT],
+    });
+
+    let state = useLivePanelState.getState();
+
+    //  do not change the state
+    state.setAvailableDetails([
+      LivePanelDetail.APPLICATION,
+      LivePanelDetail.CHAT,
+    ]);
+
+    state = useLivePanelState.getState();
+    expect(state.availableDetails).toEqual([
+      LivePanelDetail.APPLICATION,
+      LivePanelDetail.CHAT,
+    ]);
+    expect(state.currentDetail).toEqual(LivePanelDetail.CHAT);
+    expect(state.isPanelVisible).toBeTruthy();
+  });
+
+  it('without changing available details and current detail has no side effect', () => {
```

Comment:
It would be nice to read the sentence starting with the pronoun "It"

```suggestion
  it('has no side effect changing available and current details', () => {
```